rules/bicep-do-you-use-user-defined-data-types/rule.md

```diff
@@ -0,0 +1,57 @@
+---
+type: rule
+archivedreason:
+title: Bicep - Do you use User-defined Data Types?
+guid: 2a3ba816-ba40-4cfe-8193-7e151e038ac5
+uri: bicep-user-defined-data-types
+created: 2024-03-19T19:19:19.1919191Z
+authors:
+- title: Rick Su
+  url: https://ssw.com.au/people/rick-su
+related: []
+redirects: []
+
+---
+
+[User-defined data types in Bicep](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/user-defined-data-types) allow you to create custom data structures for better code organization and type safety. They enhance reusability, abstraction, and maintainability within projects.
+
+<!--endintro-->
+
+## Get Started
+
+[Bicep CLI version 0.12.X or higher](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/install) is required to use this feature.
+
+## Pain
```

Comment:
```suggestion
```

duplication of content in MS docs
pain heading doesn't add value since it includes the good example